sale_start_end_dates/models/sale.py

```diff
@@ -54,7 +54,7 @@ class SaleOrderLine(models.Model):
         states={'draft': [('readonly', False)]})
     number_of_days = fields.Integer(string='Number of Days')
     must_have_dates = fields.Boolean(
-        related='product_id.must_have_dates', readonly=True)
```

Comment:
You don't have to remove this readonly. This is correct. The problem is in product.product constraint, not here.